viewing_party/party.py

```diff
@@ -1,23 +1,189 @@
 # ------------- WAVE 1 --------------------
 
-def create_movie(title, genre, rating):
+#DONE
+import re
+
+
+def create_movie(movie_title, genre, rating):
+    dict = {}
+    
+    if movie_title is None:
+        return None
+    elif genre is None:
+        return None
+    elif rating is None:
+        return None
+    else:
+        dict["title"] = movie_title
+        dict["genre"] = genre
+        dict["rating"] = rating
+        return dict
+    pass
+
+#DONE
+def add_to_watched(user_data, movie):
+    emptyDict = {}
+    user_data["watched"].append(emptyDict)
+    user_data["watched"][0]["title"] = movie["title"]
+    user_data["watched"][0]["genre"] = movie["genre"]
+    user_data["watched"][0]["rating"] = movie["rating"]
+    return user_data
+    pass
+
+#DONE
+def add_to_watchlist(user_data, movie):
+    emptyDict = {}
+    user_data["watchlist"].append(emptyDict)
+    user_data["watchlist"][0]["title"] = movie["title"]
+    user_data["watchlist"][0]["genre"] = movie["genre"]
+    user_data["watchlist"][0]["rating"] = movie["rating"]
+    return user_data
+    pass
+
+#DONE
+def watch_movie(janes_data, movie_title):
+    for i in range(len(janes_data["watchlist"])):
+        if janes_data["watchlist"][i]["title"] == movie_title:
+            removed = janes_data["watchlist"].pop(i)
+            janes_data["watched"].append(removed)
+    return janes_data

```

Comment:
Great work in avoiding the dangers of modifying a data structure while iterating by using a ranged loop. 

One way we can improve the readability of our code is by using descriptive variable names. In this case, `i` is a very generic name that could be changed to `movie_id` to be a little more descriptive of what each index actually represents.

Reply:
Another approach is making a copy to traverse and modify the original data structure.

```python
    for movie in copy_of_user_data["watchlist"]:
        if movie_title == copy_of_user_data["watchlist"]["title"]:
            user_data["watchlist"].remove(movie)
            user_data["watched"].append(movie)
    return user_data       
```